src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -3884,6 +3905,66 @@ private boolean atEndOfFile() {
     return result;
   }
 
+  private Position getCurrentPosition() {
+    // 0-indexed
+    int lineNumber = this.getLineNumber() - 1;
+    return new Position(lineNumber, this.currentIndex);
+  }
+
+  private Range rangeToHere(final Position start) {
+    return new Range(start != null ? start : this.getCurrentPosition(), this.getCurrentPosition());
+  }
+
+  private static Range makeInlineRange(final Position start, final int offset) {
+    Position end = new Position(start.getLine(), start.getCharacter() + offset);
+
+    return offset >= 0 ? new Range(start, end) : new Range(end, start);
+  }
+
+  private static Range mergeRanges(final Range start, final Range end) {
+    if (end == null
+        || start.getStart().getLine() > end.getEnd().getLine()
```

Comment:
Can you instead use Positions.isBefore(end, start)?

Reply:
Well I'll be... I'm pleasantly surprised to learn of these utility methods...

Reply:
As far as I can tell, it's just the one :(

https://github.com/eclipse/lsp4j/blob/main/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/util/Positions.java

Reply:
well, this at least made me notice Ranges.containsRange and Ranges.containsPosition, which may come in handy in the future...